components/server/src/auth/resource-access.spec.ts

```diff
@@ -833,6 +833,76 @@ class TestResourceAccess {
                 teamRole: "owner",
                 expectation: true,
             },
+            // prebuild
+            {
+                name: "prebuild get owner",
+                resourceKind: "prebuild",
+                workspaceType: "prebuild",
+                isOwner: true,
+                teamRole: undefined,
+                expectation: true,
+            },
+            {
+                name: "prebuild get other",
+                resourceKind: "prebuild",
+                workspaceType: "prebuild",
+                isOwner: false,
+                teamRole: undefined,
+                expectation: false,
```

Comment:
@geropl, out of curiosity, what the expected effect of this change for a non-team-member to open a workspace on a configured repository? e.g. is this rule also enforced on workspace startup?

Reply:
This PR does just introduce the GuardedPrebuild, but it is not used yet.

A follow up PR will introduce prebuild APIs (get/findByWorkspaceId), and it felt wrong to re-/ab-use workspace or workspaceinstance for it.